map_test.go

```diff
@@ -31,6 +32,12 @@ var (
 	}
 )
 
+type bpfCgroupStorageKey struct {
```

Comment:
Is this a necessity? Feels a bit out of place declaring this in the test suite. If this key is a requirement for being able to write to a per-CPU-Cgroup map, shouldn't this be exposed from the package instead?

Reply:
We can just move it to the test itself in that case. I'm against adding this as an exported API:

- Seems like newer kernels accept just uint64
- Adding key types for every map is going to get unwieldy
- Adds a custom exported type for a niche (I'm assuming?) use case

Reply:
If I understand correctly (and that's a big "if"), when used, the attach type field allows multiple eBPF programs of different type to have dedicated cgroup local storage. On the other hand, without this field, the eBPF programs would be forced to share their storage area.

If so, I'd suggest a helper to create the key from a cgroup path and either a prog or a link. Would this be more acceptable ? If so, I may propose a patch in a further PR later in August (I have some holidays coming).

Reply:
Let's leave this as is for now. I do think the library should provide a concrete type for assembling composite map keys to work with sharded maps like these. We currently don't have the right interfaces for interacting with maps that would make this user-friendly. I have some ideas on how to improve this, but not to be discussed here.